odoo/addons/base/tests/test_res_lang.py

```diff
@@ -40,3 +42,63 @@ def test_00_intersperse(self):
         assert intersperse("abc12", [3], '.') == ('abc12', 0)
         assert intersperse("abc12", [2], '.') == ('abc12', 0)
         assert intersperse("abc12", [1], '.') == ('abc1.2', 1)
+
+    def test_res_lang_decimal_point_preview(self):
+        lang_0 = self.env['res.lang'].create({
+            'name': 'Test Lang',
+            'iso_code': 'TL',
+            'code': 'TL',
+        })
+        self.assertEqual(lang_0.decimal_point_preview, '99.00')
+
+        lang_0.write({'decimal_point': ','})
+
+        self.assertEqual(lang_0.decimal_point_preview, '99,00')
+
+    def test_res_lang_thousand_sep_preview(self):
+        lang_0 = self.env['res.lang'].create({
+            'name': 'Test Lang',
+            'iso_code': 'TL',
+            'code': 'TL',
+            'grouping': '[3,0]',
+        })
+
+        self.assertEqual(lang_0.thousand_sep_preview, '999,999,999')
+
+        lang_0.write({'thousands_sep': '.'})
+
+        self.assertEqual(lang_0.thousand_sep_preview, '999.999.999')
+
+    def test_res_lang_date_format_preview(self):
+        class MockedDatetime(datetime):
+            @classmethod
+            def now(cls):
+                return datetime(2020, 1, 1, 16, 0, 0)
+
+        with patch('odoo.fields.Datetime', MockedDatetime):
```

Comment:
Same here, could you use freeze_time?

You should be able to use it like this:

```
@freeze_time("2020-01-01 16:00:00")
def test_res_lang_date_format_preview(self):
    ...
```